modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/freelist/PagesList.java

```diff
@@ -1379,14 +1377,7 @@ private byte getFlag(@Nullable IoVersions<?> initIoVers) {
         if (initIoVers != null) {
             PageIo pageIo = initIoVers.latest();
 
-            switch (pageIo.getType()) {
-                case T_META:
-                case T_DATA:
-                case T_DATA_PART:
-                    return FLAG_DATA;
-                default:
-                    return defaultPageFlag;
-            }
+            return pageIo.getFlag() == FLAG_DATA ? FLAG_DATA : defaultPageFlag;
```

Comment:
I'm pretty sure that you can return `pageIo.getFlag()`, we don't have INDEX flag enymore